### PR TITLE
Breadcrumbs Types

### LINKED ISSUE #553: Refactor Navigation Breadcrumbs to Typescript
Create typings for [Navigation Breadcrumbs](https://eds.equinor.com/components/navigation-breadcrumbs/)

### PR DESCRIPTION
**Pull request by pomfrida**
Resolves #553

### CODE REVIEW

File: libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
-// @ts-nocheck
 import React, { forwardRef } from 'react'
-import PropTypes from 'prop-types'
 import styled, { css } from 'styled-components'
 import { Typography } from '../Typography'
 import { Tooltip } from '../Tooltip'
 import { breadcrumbs as tokens } from './Breadcrumbs.tokens'
 
-const StyleTypography = styled(Typography)`
+type StyledProps = Pick<Props, 'maxWidth'>
```
Comment (vnys) [inline libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx:7]: `Pick` is one of those things I didn’t know about, that we should probably use in more components…

File: libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx [hunk 2]
```diff
@@ -27,13 +27,25 @@ const StyleTypography = styled(Typography)`
   ${({ maxWidth }) => css({ maxWidth })}
 `
 
-export const Breadcrumb = forwardRef(function Breadcrumb(
-  { className, children, maxWidth, ...rest },
+type Props = {
+  /*
```
Comment (vnys) [inline libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx:31]: why a blank line here?

File: libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx [hunk 3]
```diff
@@ -38,13 +36,20 @@ const Collapsed = styled(Typography)`
...
 
-export const Breadcrumbs = forwardRef(function Breadcrumbs(
-  { className, children, collapse, ...rest },
+type Props = {
+  /*
+   * Collapses the list of breadcrumbs so that only the first
+   * and last breadcrumb will be shown, with an ellipsis in between.
+   */
+  collapse?: boolean
+} & JSX.IntrinsicElements['nav']
```
Comment (vnys) [inline libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx:45]: This is also something we do differently from component to component… `React.HTMLAttributes<HTMLNavElement> `vs. `JSX.IntrinsicElements['nav']`

**Review by vnys — APPROVED** [review]
LGTM 👍 Some extra lines in comments that could be removed, but nothing serious.

File: libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx (hunk 3 above)
Comment (pomfrida) [reply libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx:45]: @mimarz and I discussed this yesterday, if I remember correctly the difference was that `JSX.IntrinsicElements['nav']` included more props?

File: libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx (hunk 1 above)
Comment (mimarz) [reply libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx:7]: Recommend having a look at the Typescript [utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html)  :)

File: libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx (hunk 3 above)
Comment (mimarz) [reply libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx:45]: @vnys Funfact, [nav](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav) only uses the `HTMLElement` dom interface

Comment (vnys) [reply libraries/core-react/src/Breadcrumbs/Breadcrumbs.tsx:45]: > @vnys Funfact, [nav](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nav) only uses the `HTMLElement` dom interface

Noted 😉

File: libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx (hunk 1 above)
Comment (pomfrida) [reply libraries/core-react/src/Breadcrumbs/Breadcrumb.tsx:7]: It's pretty neat! 😄  @mimarz also taught me `Omit<Type, Keys>` from [Typescript Utility Types](https://www.typescriptlang.org/docs/handbook/utility-types.html) where you can remove `Keys` from `Type` and re-use it

**Pull request merged**